app/test_main.py

```diff
@@ -1 +1,99 @@
-# write your code here
+import pytest
+from unittest import mock
+from requests.exceptions import (
+    MissingSchema,
+    ConnectionError,
+    InvalidURL
+)
+
+from app.main import can_access_google_page
+
+
+@mock.patch("app.main.valid_google_url")
+def test_valid_google_url_works_in_can_access_google_page(
+        mocked_valid_url: any
+) -> None:
+    can_access_google_page("http://google.com")
+
+    mocked_valid_url.assert_called_once()
+
+
+@mock.patch("app.main.has_internet_connection")
+def test_has_internet_connection_works_in_can_access_google_page(
+        mock_internet_connection: any
+) -> None:
+    can_access_google_page("http://google.com")
+
+    mock_internet_connection.assert_called_once()
+
+
+@mock.patch("app.main.has_internet_connection")
+@pytest.mark.parametrize(
+    "initial_url,initial_hour,expected_result",
+    [
+        pytest.param(
+            "http://google.com",
+            True,
+            "Accessible",
+            id="check valid url and right hour"
+        ),
+        pytest.param(
+            "http://google.com",
+            False,
+            "Not accessible",
+            id="hour has to be from 6 to 22"
+        ),
+        (
+            "https://translate.google.com/",
+            False,
+            "Not accessible"
+        )
+    ]
+)
+def test_can_access_google_page(
+        mocked_time: any,
+        initial_url: str,
+        initial_hour: int,
+        expected_result: bool
```

Comment:
Please correct the type annotations. mocked_time is a MagickMock object, initial_hour is a boolean value and the expected_result is a string